src/apis/social_network/controllers/posts.controller.js

```diff
@@ -31,5 +31,86 @@ module.exports = {
       err.func = err.func || 'getPostsForTimeline'
       errorHandlingService.handleErrorInRequest(req, res, err)
     }
+  },
+
+  getPostData: async function(req, res) {
+    const userId = req.api.userId
+    const postId = req.params.post_id
+
+    try {
+      const postInPrivateGroup = await postService.postBelongsToPrivateGroup(postId)
+
+      if (postInPrivateGroup.group_private) {
+        // User not authenticated
+        if (userId === undefined) {
+          return res.status(401).finish({
+            code: 1,
+            messages: ['Unauthenticated. User must authenticate to get the requested response.']
+          })
+        }
+        // User authenticated
+        const userIsMember = await postService.userBelongsToGroup(userId, postInPrivateGroup.group_id)
+
+        if (userIsMember) {
+          const post = await postService.getPostDataUserAuth(postId, userId)
+          if (post.referenced_post_id != null) {
+            post.referenced_post = await postService.getBasePostDataUserAuth(post.referenced_post_id, userId)
+          }
+          post.referenced_post_id = undefined
+          return res.finish({
+            code: 0,
+            messages: ['Done'],
+            data: post
+          })
+        } else {
+          return res.status(403).finish({
+            code: 2,
+            messages: [`Forbidden. The requested post belongs to a private group to which the user requesting doesn't belong.`]
+          })
+        }
+      } 
+
+      // User not authenticated
+      if (userId === undefined) {
+        const post = await postService.getPostData(postId)
+        if (!post) {
+          return res.status(404).finish({
+            code: 3,
+            messages: ['Post not found']
+          })
+        }
+        if (post.referenced_post_id != null) {
+          post.referenced_post = await postService.getBasePostData(post.referenced_post_id)
+        }
+        post.referenced_post_id = undefined
+        return res.finish({
+          code: 0,
+          messages: ['Done'],
+          data: post
+        })
+      }
+      // User authenticated
+      const post = await postService.getPostDataUserAuth(postId, userId)
+      if (!post) {
+        return res.status(404).finish({
+          code: 3,
+          messages: ['Post not found']
+        })
+      }
+      if (post.referenced_post_id != null) {
+        post.referenced_post = await postService.getBasePostDataUserAuth(post.referenced_post_id, userId)
+      }
+      post.referenced_post_id = undefined
+      return res.finish({
+        code: 0,
+        messages: ['Done'],
+        data: post
+      })
```

Comment:
Due to changes proposed in `getBasePostDataUserAuth` and `getPostDataUserAuth`, this part of the code could decrease to the half.

For messages, in this case you could use general messages:

* e404: 'Resource not found'

and for a success message, we could start to use:

* c200: 'Done'

This message could be in a field called `success_messages`

Add it in etc/messages.json